src/ispyb/__init__.py

```diff
@@ -1,44 +1,59 @@
 import configparser
 import logging
+import os
 
 __version__ = "6.0.1"
 
 _log = logging.getLogger("ispyb")
 
 
-def open(configuration_file):
-    """Create an ISPyB connection using settings from a configuration file.
-    This can be used either as a function call or as a context manager.
+def open(credentials=None):
+    """Create an ISPyB connection.
```

Comment:
Strictly this is an API change, so I have mixed feelings about it. On the other hand I don't believe anyone sane referred to this by name, so it is probably fine.

Reply:
I changed it to be consistent with the recently-added [`ispyb.sqlalchemy.session`](https://github.com/DiamondLightSource/ispyb-api/blob/4d11686ccf872304276824ecb06e6b9cead0869c/src/ispyb/sqlalchemy/__init__.py#L20), but happy to switch the name back to `configuration_file` if you think that would be preferable.

Reply:
Added a commit to allow both plus warning. I think this should be a theoretical case, but you never know :shrug: